datasets/fr/amf_regulatory_sanctions/crawler.py

```diff
@@ -0,0 +1,161 @@
+from typing import Any, Generator, Optional, Tuple, Union
+
+from normality import collapse_spaces
+from zavod import Context, helpers as h
+import re
+from html import unescape
+from urllib.parse import urljoin
+from lxml import html
+
+
+CLEAN_ENTITY = re.compile(r"(<br />\r\n| et |;)", re.IGNORECASE)
+BASE_URL = "https://www.amf-france.org"
+
+
+def parse_json(context: Context) -> Generator[dict, None, None]:
+    response = context.fetch_json(context.data_url)
+    if "data" not in response:
+        context.log.info("No data available.")
+        return
+
+    context.log.info(
+        f"Fetched {len(response['data'])} results."
+    )  # Log the number of results
+    for item in response["data"]:
+        yield item
```

Comment:
mainly a nitpick, but I'd call this "fetch_items". What it's doing is
- fetching json
- parsing json from text to python data structures
- yielding items

a function called "parse_json" belongs in the `json` module - this function is so much more, and readers would benefit from knowing it's a high level abstraction to fetch stuff and not just a data format parser